SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerClusterizer.cc

```diff
@@ -0,0 +1,98 @@
+#include "SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerClusterizer.h"
+
+#include "Geometry/CommonDetUnit/interface/GeomDetUnit.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/SiPixelDigi/interface/PixelDigi.h"
+
+#include "FWCore/PluginManager/interface/ModuleDef.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include <vector>
+
+namespace cms {
+
+    /*
+     * Initialise the producer
+     */ 
+
+    Phase2TrackerClusterizer::Phase2TrackerClusterizer(edm::ParameterSet const& conf) :
+        conf_(conf),
+        src_(conf.getParameter< edm::InputTag >("src")),
+        maxClusterSize_(conf.getParameter< unsigned int >("maxClusterSize")),
+        maxNumberClusters_(conf.getParameter< unsigned int >("maxNumberClusters")) {
+            // Objects that will be produced
+            produces< Phase2TrackerCluster1DCollectionNew >(); 
+            // 
+            clusterizer_  = new Phase2TrackerClusterizerAlgorithm(maxClusterSize_, maxNumberClusters_);
```

Comment:
This memory is never released. Either change clusterizer_ to a std::unique_ptr or delete it in the destructor.